project/project.go

```diff
@@ -70,25 +79,34 @@ func (proj *Project) MergeFrom(mergeProj Project) error {
 			return nil
 		}
 
-		fileContent, err := ioutil.ReadFile(filePath)
+		fileReader, err := os.Open(filePath)
 		if err != nil {
 			return fmt.Errorf("failed to read '%s'", filePath)
 		}
 
-		if !proj.ShouldMerge(filePath, fileContent) {
+		if !proj.ShouldMerge(filePath, fileReader) {
 			return nil
 		}
 
+		fileReader.Close()
+
 		// Merge the file
 
+		srcFile, err := os.Open(filePath)
+		if err != nil {
+			return fmt.Errorf("failed to read '%s'", filePath)
+		}
+		defer srcFile.Close()
+
 		destFilePath := strings.Replace(filePath, mergeProj.path, proj.path, 1)
 
 		destFile, err := os.Create(destFilePath)
```

Comment:
Again, bubbling the os.Create operations outside of the anonymous functions may make it easier to test logic in this walk function.